### PR TITLE
Feat/push notif trigger endpoint

### PR DESCRIPTION
**Pull request by kwajiehao**
## Overview

We want to be able to send push notifications to remind people to submit their health reports (or location reports) as and when we need to. With our APIs, sending push notifications requires us to:
1. Obtain a list of (latest) quarantine order IDs from a list of contact numbers
2. Retrieve the current push notification ID for each quarantine order
3. Use the `NotificationService` to send push notifications using each of these push notification IDs

Currently, this process is being done using manual scripts. This PR introduces an endpoint which does the same thing.

### Specify a channel to send the push notification for Android
From preliminary tests, notifications weren't showing up prominently on Android. Our mobile app had several special channels configured to receive push notifications, but since we didn't configure those channels in the backend, the notifications being sent were not displayed prominently. This PR introduces code and env vars which select the proper channel to send push notifications to for Android.

### Add an endpoint for sending push notifications given a list of phone numbers
This PR introduces a `POST /push-notifications/send` endpoint which accepts an array of ~phone numbers~ quarantine order IDs and a type, which determines whether the push notification sent is for the health report or the location report. Posting to this endpoint will trigger push notifications to the quarantine orders associated with the phone numbers submitted.

### Add an endpoint which sends SMS notifications given a list of phone numbers
This PR also introduces a `POST /push-notifications/send-sms` endpoint which accepts an array of phone numbers and a type. The type is used to determine the content of the SMS sent: `Please open the Homer app to submit your ${type} report.`

### Add an endpoint which sends push notifications if possible, otherwise default to SMS
Finally, the endpoint `POST /push-notifications/push-and-send` sends push notifications given an array of ~phone numbers~ quarantine order IDs and a type. If push notification tokens can be found for the quarantine orders, then a push notification is sent. Otherwise, a SMS notification is sent which asks the user to open the app and submit a report.

### Misc: Add a `GET /orders/all/latest ` endpoint
This endpoint retrieves the latest quarantine order for each phone number given an array of phone numbers. ~It helps us complete step 1 of the push notification sending process above~. 

## Note
~Unfortunately, since we do not have a toggle between staging and production, and the current Homer app points to the production setup, I was unable to test this endpoint, specifically the pushing of the notification itself (line 164 to 172 of `app/services/NotificationService/index.js`) since I am unable to obtain a valid push notification ID without going on production. I have however tested everything up till that point by creating a dummy push notification ID on the staging DB.~ This PR was tested on the production DB with Liwei and Kaiwen's Android phones.

### CODE REVIEW

File: app/routes/orders.js [hunk 1]
```diff
@@ -142,7 +147,23 @@ async function bulkCreateQuarantineOrder(req, res) {
...
-async function updateOrder(req, res) {
+// This route function retrieves all latest quarantine orders for each phone number
+// given an array of numbers
+async function getAllLatestOrdersByNumber(req, res) {
+  try {
+    const validRequest = ajv.validate(getAllLatestOrdersSchema, req);
+    if (!validRequest) throw new InputSchemaValidationError(JSON.stringify(ajv.errors));
+
+    const { contact_number_array: contactNumberArr } = req.body;
+    const results = await QuarantineOrderService.getLatestOrdersByContactNumberArr(contactNumberArr);
```
Comment (christabelneo) [inline app/routes/orders.js:158]: can just map with `getLatestOrderByContactNumber`

**Comment by kwajiehao** [comment]
> lgtm - we can refactor tomorrow and also add a new feature to make `contact_array` optional; if `contact_array` is undefined, send the PN/SMS out to all active quarantine orders.

Refactored the push notification and sms notification route functions to use components which specifically send out bulk push notifications / sms notifications. The optional argument feature will be completed in a separate PR

File: app/routes/pushNotifications.js [hunk 2]
```diff
@@ -126,9 +134,266 @@ async function handlePingHealthReport(request, response) {
...
+    // Retrieve quarantine orders
+    const { qo_array: quarantineOrders, type } = req.body;
+    const {
+      missingPushNotificationOrders,
+      unsuccessfulPushNotifications,
+      results: pushResults,
+    } = await sendPushNotifications(quarantineOrders, type);
+
+    // Send SMS
+    const contactNumberArr = missingPushNotificationOrders.map((order) => order.contact_number);
```
Comment (prestonlimlianjie) [inline app/routes/pushNotifications.js:368]: Throws an error if `missingPushNotificationOrders` is empty

File: app/routes/pushNotifications.js [hunk 3]
```diff
@@ -126,9 +134,266 @@ async function handlePingHealthReport(request, response) {
...
+      : `POST /push-notifications/push-and-send unhandled server error: ${JSON.stringify(err)}`);
+    return res.status(err.status ? err.status : 500).send(`${err.name}: ${err.message}`);
+  }
+}
+
 router.post('/register', handleRegister);
 router.post('/location', handlePingLocation);
 router.post('/photo', handlePingPhoto);
 router.post('/health-report', handlePingHealthReport);
+router.post('/send', sendPushNotificationsRoute);
```
Comment (prestonlimlianjie) [inline app/routes/pushNotifications.js:395]: I think we decided to rename these 3 endpoints `/push`, `/sms`, and `/push-and-sms`

**Pull request merged**